### PR TITLE
toolchain for Cheyenne intel

### PR DESCRIPTION
**Pull request by mmiesch**
Accompanies  https://github.com/JCSDA/jedi-stack/pull/138

This adds a toolchain for building jedi on Cheyenne with Intel, with flags to enable C++-14 support

### CODE REVIEW

**Comment by mmiesch** [comment]
also a toolchain for Orion

File: cmake/Toolchains/jcsda-Cheyenne-Intel.cmake [hunk 1]
```diff
@@ -0,0 +1,12 @@
+####################################################################
+# Compiler flags
+####################################################################
+
+set( CMAKE_CXX_FLAGS   "-gxx-name=/glade/u/apps/ch/opt/gnu/9.1.0/bin/g++ -Wl,-rpath,/glade/u/apps/ch/opt/gnu/9.1.0/lib64" CACHE STRING "gnu headers for c++14" FORCE )
```
Comment (markjolah) [inline cmake/Toolchains/jcsda-Cheyenne-Intel.cmake:5]: Hmm... the `-Wl,-rpath` is an option being sent specifically to the linker.  I would expect only the `CMAKE_LINKER_FLAGS` would require it?

File: cmake/Toolchains/jcsda-Cheyenne-Intel.cmake [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+set( CMAKE_LINKER_FLAGS   "-gxx-name=/glade/u/apps/ch/opt/gnu/9.1.0/bin/g++ -Wl,-rpath,/glade/u/apps/ch/opt/gnu/9.1.0/lib64" CACHE STRING "gnu headers for c++14" FORCE )
```
Comment (markjolah) [inline cmake/Toolchains/jcsda-Cheyenne-Intel.cmake:6]: I can't find documentation on `CMAKE_LINKER_FLAGS` it looks like you need to be setting both `CMAKE_EXE_LINKER_FLAGS` and `CMAKE_SHARED_LINKER_FLAGS`, and then can remove the setting in `CMAKE_CXX_FLAGS`.

**Review by rickgrubin — APPROVED** [review]
LGTM

**Comment by mmiesch** [comment]
Ok - I made the requested changes - I haven't tested them yet but let's get another approval so we can merge this - the jedi docs are out of date with Orion

**Review by rhoneyager — APPROVED** [review]

**Pull request merged**